Added Loading Screen with Success/Error Feedback for Registration and Login

DerekAyala opened the pull request:
# Description

This issue involves improving the user experience during the registration and login process by adding a loading screen. The loading screen should provide feedback on the success or failure of the action, displaying different icons and messages accordingly.

# changes made:
- Created a new component called "LoadingModal" to serve as a loading screen during the registration and login process.
- Implemented the necessary logic to display the loading screen when a registration or login request is in progress.
- Designed the loading screen to show a spinner or loading animation while the action is being processed.
- Added functionality to display a success or error icon, along with an appropriate message, upon completion of the registration or login process.
- Integrated the LoadingModal component into the registration and login flows to provide a seamless and responsive user experience with feedback on the outcome of the action.
# Testing
## Email sign in
![email sign in](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-ghostbusters/assets/89038565/0c7c40ef-8386-4981-9aac-2438e92e08b0)
![Screenshot_1688695149](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-ghostbusters/assets/89038565/ecd7f5e4-cde1-4a25-b67a-fa97998c6d6d)
![Screenshot_1688695150](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-ghostbusters/assets/89038565/1c2979b5-5b26-4685-984d-5388fb39ea9d)

## Email Sign up error
![Screenshot_1688622227](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-ghostbusters/assets/89038565/d82963fa-3f6d-4ddd-8bcd-aae08f2da6c4)
![Screenshot_1688622238](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-ghostbusters/assets/89038565/fff30250-06cd-490b-9a0f-1bd63da47de9)
![Screenshot_1688622232](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-ghostbusters/assets/89038565/e8c48486-6f66-4a39-9c8e-0ca8dce733f0)
## Email Sign up
![Screenshot_1688622317](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-ghostbusters/assets/89038565/f6d4a293-a500-455b-bd6a-a55688100ebd)

Commit:
Added Modal Component for Loading Screen during Registration and Login

- Created a new component called "LoadingModal" to serve as a loading screen during the registration and login process.
- Implemented the necessary logic to show the modal when a registration or login request is in progress.
- Designed the modal to display a spinner or a progress indicator to indicate that the system is processing the request.
- Integrated the LoadingModal component into the registration and login flows to provide a seamless and responsive user experience.
- Tested the functionality to ensure that the loading modal appears and disappears at the appropriate times during the registration and login processes.

Co-authored-by: Eduardoespl <eduardoesp3317@gmail.com>

Commit:
Loading Screen during Google Sign-In

-Implemented the necessary logic to show the modal when a registration, login, or Google Sign-In request is in progress.

Co-authored-by: Eduardoespl <eduardoesp3317@gmail.com>

Commit:
Added Modal Component with Icon and Confirmation Message Props

- Updated the LoadingModal component to include a prop for the error icon name. This prop allows displaying an error icon when there is an error during the registration or login process.

Co-authored-by: Eduardoespl <eduardoesp3317@gmail.com>

Tole15 commented:
Hey, buen trabajo 👍

YilKanda commented on AppVuelos/src/screens/register.tsx:
```diff
@@ -39,25 +44,49 @@ function Register(): JSX.Element {
-      setRegisterError('Email in use. Use a different email')
-    })
+      .then(userCredential => {
+        console.log(userCredential);
```
Don't leave debugging code like console.log()

YilKanda commented on AppVuelos/src/screens/register.tsx:
```diff
@@ -39,25 +44,49 @@ function Register(): JSX.Element {
+        }, 2000);
+      })
+      .catch(error => {
+        console.log(error);
```
Don't leave debugging code like console.log()

YilKanda commented on AppVuelos/src/screens/register.tsx:
```diff
@@ -39,25 +44,49 @@ function Register(): JSX.Element {
   const handleButtonClick = async () => {
+    setvisible(true);
     const formData = getFormData(firstName, email, password);
     console.log(formData);
```
Don't leave debugging code like console.log()

Eduardoespl committed:
Comment the console.log in the files

Modify files from screens and hooks folders to comment the console.log, this is to avoid the console.log in the app, we dont delete the console.log because we need them to continue developing.

Co-authored-by: DerekAyala <derek.ayala12345@hotmail.com>
